react/README.md

```diff
@@ -1,50 +1,120 @@
-# React + TypeScript + Vite
+# React GridStack Wrapper Demo
 
-This template provides a minimal setup to get React working in Vite with HMR and some ESLint rules.
+A React wrapper component for GridStack that provides better TypeScript support and React integration experience.
 
-Currently, two official plugins are available:
+## TODO
 
-- [@vitejs/plugin-react](https://github.com/vitejs/vite-plugin-react/blob/main/packages/plugin-react/README.md) uses [Babel](https://babeljs.io/) for Fast Refresh
-- [@vitejs/plugin-react-swc](https://github.com/vitejs/vite-plugin-react-swc) uses [SWC](https://swc.rs/) for Fast Refresh
+- [x] Component mapping
+- [x] SubGrid support
+- [ ] Save and restore layout
+- [ ] Publish to npm
 
-## Expanding the ESLint configuration
+## Basic Usage
 
-If you are developing a production application, we recommend updating the configuration to enable type aware lint rules:
+This is not an npm package, it's just a demo project. Please copy the relevant code to your project to use it.
 
-- Configure the top-level `parserOptions` property like this:
+```tsx
+import {
+  GridStackProvider,
+  GridStackRender,
+  GridStackRenderProvider,
+} from "path/to/lib";
+import "gridstack/dist/gridstack.css";
+import "gridstack/dist/gridstack-extra.css";
+import "path/to/demo.css";
 
-```js
-export default tseslint.config({
-  languageOptions: {
-    // other options...
-    parserOptions: {
-      project: ['./tsconfig.node.json', './tsconfig.app.json'],
-      tsconfigRootDir: import.meta.dirname,
+function Text({ content }: { content: string }) {
+  return <div>{content}</div>;
+}
+
+const COMPONENT_MAP = {
+  Text,
+  // ... other components
+};
+
+// Grid options
+const gridOptions = {
+  acceptWidgets: true,
+  margin: 8,
+  cellHeight: 50,
+  children: [
+    {
+      id: "item1",
+      h: 2,
+      w: 2,
+      content: JSON.stringify({
```

Comment:
is there a good reason to have a json structure here instead of just 
content: 'Item 1'
I suppose you can make it other thing than name:'text' (should be type) but real apps are likely to use actual components looked up by unique id or something (see angular wrapper 'selector' field)

Reply:
Good idea. I didn't think of doing that at the time. Since initially I wanted to store react component information on the same set of data structures, I'll try to do it your way, and may need to maintain an additional copy of the mapping relationship.

Reply:
great, and also try to use the same class names and fields as I have done for Angular wrapper if possible ('selector' is an angular thing since that is the field we use, so pick the native React equivalent).

Maybe @damien-schneider and @GowthamTG can also chime on this so maybe between the 3 of you we can get an official React wrapper as well. I appreciate the help as I don't know React.
Also demo should be more like two.html (maybe still with nested grid) so we make sure dragging between grids also works. The key is to let Gridstack handle the grid/container creation/reparenting/layout and have React get called back when content (or in the Angular case the grid and gridItem as parents also have to be Angular based) needs to be created using native platform way.